### PR TITLE
Provide a @ConfigDocEnum annotation to enforce hyphenation of enum

### LINKED ISSUE #42514: Enum property values not hyphenated anymore in the configuration metadata
### Describe the bug

We have a config property in Quarkus CXF defined as follows:

```java
    @WithDefault("Keep-Alive")
    @WithConverter(ConnectionTypeConverter.class)
    public ConnectionType connection();
```

Before the recent refactoring of `quarkus-extension-processor`, the metadata was produced with hyphenated enum values `target/asciidoc/generated/config/all-configuration-roots-generated-doc/io.quarkiverse.cxf.CxfConfig`:

```json
  {
    "configDocKey": {
      "type": "org.apache.cxf.transports.http.configuration.ConnectionType",
      "key": "quarkus.cxf.client.\"client-name\".connection",
      "additionalKeys": [],
      "configDoc": "The connection disposition. If close the connection to the server is closed after each request/response dialog. If\nKeep-Alive the client requests the server to keep the connection open, and if the server honors the keep alive request,\nthe connection is reused. Many servers and proxies do not honor keep-alive requests.",
      "withinAMap": true,
      "defaultValue": "keep-alive",
      "javaDocSiteLink": "",
      "docMapKey": "connection",
      "configPhase": "RUN_TIME",
      "acceptedValues": [
        "`close`",
        "`keep-alive`"
      ],
      "optional": false,
      "list": false,
      "passThroughMap": false,
      "withinAConfigGroup": true,
      "topLevelGrouping": "quarkus.cxf",
      "since": "2.2.3",
      "environmentVariable": "QUARKUS_CXF_CLIENT__CLIENT_NAME__CONNECTION",
      "enum": true
    }
  },
```

After the refactoring, the enum values in `target/quarkus-config-doc/quarkus-config-model.yaml` are present verbatim:

```yaml
  - !<io.quarkus.annotation.processor.documentation.config.model.ConfigProperty>
    enum: true
    sourceClass: "io.quarkiverse.cxf.CxfClientConfig"
    sourceName: "connection"
    path: "quarkus.cxf.client.\"client-name\".connection"
    type: "org.apache.cxf.transports.http.configuration.ConnectionType"
    phase: "RUN_TIME"
    environmentVariable: "QUARKUS_CXF_CLIENT__CLIENT_NAME__CONNECTION"
    typeDescription: "ConnectionType"
    withinMap: true
    converted: true
    enumAcceptedValues:
      qualifiedName: "org.apache.cxf.transports.http.configuration.ConnectionType"
      values:
        CLOSE:
          configValue: "CLOSE"
        KEEP_ALIVE:
          configValue: "KEEP_ALIVE"
    defaultValue: "Keep-Alive"
```

### Expected behavior

The enum values should be produced hyphenated, same as before quarkus-extension-processor refactoring.

I am aware of the following note in ConfigResolver

https://github.com/quarkusio/quarkus/blob/baf77db720ab5794a1b152c87a01df3c291f864e/core/processor/src/main/java/io/quarkus/annotation/processor/documentation/config/resolver/ConfigResolver.java#L155-L156

However my example proves that the rule to decide about hyphenation must have been complexer than just "if the property has a converter"

### PR DESCRIPTION
**Pull request by gsmet**
Also refactor things a bit to centralize the handling of the common annotations.

Fixes #42514

### CODE REVIEW

**Comment by gsmet** [comment]
@ppalaga I haven't tested it. I let you have a look and tell me if it's good enough for you.

File: core/runtime/src/main/java/io/quarkus/runtime/annotations/ConfigDocEnum.java [hunk 1]
```diff
@@ -0,0 +1,23 @@
...
+ */
+@Retention(RUNTIME)
+@Target({ FIELD })
+@Documented
+public @interface ConfigDocEnum {
+
+    /**
+     * This can be used to enforce hyphenating the enum values even if a converter is present.
+     */
+    boolean enforceHyphenateValues();
```
Comment (gastaldi) [inline core/runtime/src/main/java/io/quarkus/runtime/annotations/ConfigDocEnum.java:21]: Should it have a default value?

Comment (gsmet) [reply core/runtime/src/main/java/io/quarkus/runtime/annotations/ConfigDocEnum.java:21]: Ah yes, of course, will fix.

Comment (gsmet) [reply core/runtime/src/main/java/io/quarkus/runtime/annotations/ConfigDocEnum.java]: Done, thanks!

**Review by gastaldi — APPROVED** [review]

File: core/runtime/src/main/java/io/quarkus/runtime/annotations/ConfigDocEnum.java [hunk 2]
```diff
@@ -0,0 +1,23 @@
...
+
+import java.lang.annotation.Documented;
+import java.lang.annotation.Retention;
+import java.lang.annotation.Target;
+
+/**
+ * Provides a way to configure how an enum is handled.
+ */
+@Retention(RUNTIME)
+@Target({ FIELD })
```
Comment (ppalaga) [inline core/runtime/src/main/java/io/quarkus/runtime/annotations/ConfigDocEnum.java:14]: This should contain METHOD for sure. Not sure FIELD still makes sense?

Comment (ppalaga) [reply core/runtime/src/main/java/io/quarkus/runtime/annotations/ConfigDocEnum.java:14]: It works as expected with this change. Thanks @gsmet!

Comment (gsmet) [reply core/runtime/src/main/java/io/quarkus/runtime/annotations/ConfigDocEnum.java]: Fixed it. I apparently need some rest :). Thanks for testing!

**Pull request merged**